zeebe/util/src/test/java/io/camunda/zeebe/util/jar/ExternalJarClassLoaderTest.java

```diff
@@ -78,4 +80,67 @@ void shouldLoadResourceFiles(final @TempDir File tempDir) throws Exception {
     // then
     assertThat(storedValue).isEqualTo(testResourceValue);
   }
+
+  @Test
+  void shouldCloseWhenDefaultCloseInvoked(final @TempDir File tempDir) throws Exception {
+    // given
+    final var serviceClass = ExternalService.createUnloadedExporterClass();
+    final var jarFile = serviceClass.toJar(new File(tempDir, "service.jar"));
+    final var classLoader = ExternalJarClassLoader.ofPath(jarFile.toPath());
+
+    // when
+    classLoader.close();
+
+    // then
+    Assertions.assertThrows(
+        ClassNotFoundException.class, () -> classLoader.loadClass(ExternalService.CLASS_NAME));
+  }
+
+  @Test
+  void shouldCloseWhenVerboseTrueCloseInvoked(final @TempDir File tempDir) throws Exception {
+    // given
+    final var serviceClass = ExternalService.createUnloadedExporterClass();
+    final var jarFile = serviceClass.toJar(new File(tempDir, "service.jar"));
+    final var classLoader = ExternalJarClassLoader.ofPath(jarFile.toPath());
+
+    // when
+    classLoader.close(true);
+
+    // then
+    Assertions.assertThrows(
+        ClassNotFoundException.class, () -> classLoader.loadClass(ExternalService.CLASS_NAME));
+  }
+
+  @Test
+  void shouldCloseWhenVerboseFalseCloseInvoked(final @TempDir File tempDir) throws Exception {
+    // given
+    final var serviceClass = ExternalService.createUnloadedExporterClass();
+    final var jarFile = serviceClass.toJar(new File(tempDir, "service.jar"));
+    final var classLoader = ExternalJarClassLoader.ofPath(jarFile.toPath());
+
+    // when
+    classLoader.close(false);
+
+    // then
+    Assertions.assertThrows(
```

Comment:
:x: Please use AssertJ for assertions like in the rest of Zeebe code :+1: You can use `assertThatCode(() -> ...)` to test exceptions.